### PR TITLE
Kyle Schulz

### PR DESCRIPTION
**Pull request by kylejschulz**
well that was fun :(

### CODE REVIEW

File: lib/exhibit.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class  Exhibit
+  attr_reader :name, :cost
+  def initialize(args)
+    @args = args
```
Comment (mikedao) [inline lib/exhibit.rb:4]: no need to save the entire thing just pull what you need

File: lib/museum.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    end
+    interested
+  end
+
+  def admit(patron)
+    @patrons << patron
+  end
+
+  def patrons_by_exhibit_interest
+    hash = Hash.new { |h, k| h[k] = [] }
```
Comment (mikedao) [inline lib/museum.rb:28]: Avoid using this syntax to set a default value.

File: lib/museum.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  def ticket_lottery_contestants(exhibit)
+    eligible = patrons_by_exhibit_interest[exhibit].select do |patron|
+      patron.poor_AF?(exhibit.cost)
+    end
+    eligible
+  end
+
+  def draw_lottery_winner(exhibit)
+    ticket_lottery_contestants(exhibit).sample
```
Comment (mikedao) [inline lib/museum.rb:48]: Draw is supposed to return a string.

File: lib/museum.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def announce_lottery_winner(exhibit)
+    if draw_lottery_winner(exhibit).nil?
+      puts "No winners for this lottery"
+    else
+      puts "#{draw_lottery_winner(exhibit)} has won the #{exhibit.name} lottery"
+    end
+  end
+end
```
Comment (mikedao) [inline lib/museum.rb:58]: Announce is supposed to return a string, not print to the screen.

**Review by mikedao — COMMENTED** [review]
Overall what you have is well done. Make sure you're reading the spec closely and you are writing code that matches what is being asked of you.

**Comment by kylejschulz** [comment]
@mikedao Thanks for the feedback. I do need to do a better job reading the interaction pattern. Is there a score on here that i'm missing by chance?

**Pull request closed without merging**